src/Symfony/Component/DependencyInjection/Compiler/FactoryReturnTypePass.php

```diff
@@ -29,6 +42,9 @@ public function process(ContainerBuilder $container)
         if (!method_exists(\ReflectionMethod::class, 'getReturnType')) {
             return;
         }
+        if (null !== $this->resolveClassPass) {
+            $this->resolveClassPassChanges = $this->resolveClassPass->getChanges();
```

Comment:
Did you forget to push the change or do I miss something? I don't see where it is reset.